system/src/system_control_internal.cpp

```diff
@@ -160,36 +160,32 @@ void SystemControl::processRequest(ctrl_request* req, ControlRequestChannel* /*
         break;
     }
     case CTRL_REQUEST_RESET: {
-        setResult(req, SYSTEM_ERROR_NONE, [](int result, void* data) {
-            HAL_Delay_Milliseconds(1000);
-            System.reset();
-        });
+        HAL_Delay_Milliseconds(1000);
+        auto res = System.reset();
+        setResult(req, res);
```

Comment:
I removed lambda functions here to collect the return code. Please let me know if there is more C++ way to do this.